File: 2107-find-unique-binary-string/find-unique-binary-string.py
class Solution:
    def findDifferentBinaryString(self, nums: List[str]) -> str:
        # start with zeros and make string different from every num
        n = len(nums)
        unique = ['0'] * n

        for i, num in enumerate(nums):
            if num[i] == unique[i]:
                unique[i] = '1'

        return ''.join(unique) 

File: 2107-find-unique-binary-string/find-unique-binary-string.java
class Solution {
    public String findDifferentBinaryString(String[] nums) {
        StringBuilder sb = new StringBuilder();
        int n = nums.length;
        for (int i = 0; i < n; i++) {
            sb.append(1 ^ nums[i].charAt(i) - '0');
        }
        return sb.toString();
    }
}